src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -18,20 +18,20 @@ internal static class CodeWriterExtensions
 
     private static readonly char[] CStyleStringLiteralEscapeChars =
     {
-            '\r',
-            '\t',
-            '\"',
-            '\'',
-            '\\',
-            '\0',
-            '\n',
-            '\u2028',
-            '\u2029',
-        };
+        '\r',
+        '\t',
+        '\"',
+        '\'',
+        '\\',
+        '\0',
+        '\n',
+        '\u2028',
+        '\u2029',
+    };
 
     public static bool IsAtBeginningOfLine(this CodeWriter writer)
```

Comment:
Meta question about this file. At a glance it seems like `ReadOnlySpan<char>` could be used. Basically I didn't see any obvious place it was stored in the heap. Seems pretty pass around, slice and append to a buffer. That would in theory open us up to future optimizations via `stackalloc`. 

Is that worth considering at this layer in Razor? Or is `stackalloc` not realistic and we should be focusing on `ROM<char>` instead? Or did I just miss cases where `ROM<char>` is necessary?

Reply:
There's a problem with using `ReadOnlySpan<char>`: Once you've got a span, you can't get back to a `ReadOnlyMemory<char>`. And, if we append a `ReadOnlySpan<char>` to a buffer, we end up duplicating the underlying string into the buffer, only to end up duplicating it again when we produce the generated code string. That's why this change uses `ReadOnlyMemory<char>`, so that we can stick them in a data structure until we're ready to produce the generated code. Because we' don't unpack the `ReadOnlyMemory<char>s` until the end, they can all keep pointing to the same string references, many of which are shared.

Reply:
Note: A future change might be to "write" `ReadOnlyMemory<char>s` that largely reference the same string of common C# language bits to further reduce heap allocation.